app/assets/javascripts/discourse/app/components/sidebar/edit-navigation-menu/categories-modal.gjs

```diff
@@ -18,126 +18,142 @@ import gt from "truth-helpers/helpers/gt";
 import includes from "truth-helpers/helpers/includes";
 import not from "truth-helpers/helpers/not";
 
+// Given a list, break into chunks starting a new chunk whenever the predicate
+// is true for an element.
+function splitWhere(elements, f) {
+  return elements.reduce((acc, el, i) => {
+    if (i === 0 || f(el)) {
+      acc.push([]);
+    }
+    acc[acc.length - 1].push(el);
+    return acc;
+  }, []);
+}
+
+function findAncestors(categories) {
+  let categoriesToCheck = categories;
+  const ancestors = [];
+
+  for (let i = 0; i < 3; i++) {
+    categoriesToCheck = categoriesToCheck
+      .map((c) => Category.findById(c.parent_category_id))
+      .filter(Boolean)
+      .uniqBy((c) => c.id);
+
+    ancestors.push(...categoriesToCheck);
+  }
+
+  return ancestors;
+}
+
 export default class extends Component {
   @service currentUser;
   @service site;
   @service siteSettings;
 
-  @tracked filter = "";
-  @tracked filteredCategoryIds;
-  @tracked onlySelected = false;
-  @tracked onlyUnselected = false;
+  @tracked filteredCategoriesGroupings = [];
+  @tracked filteredCategoryIds = [];
 
   @tracked
   selectedSidebarCategoryIds = [...this.currentUser.sidebar_category_ids];
 
-  categoryGroupings = [];
-
   constructor() {
     super(...arguments);
 
-    let categories = [...this.site.categories];
+    this.processing = false;
+    this.setFilterAndMode("", "everything");
```

Comment:
We should add a loading state here, since now we're making an ajax call to get the categories. 

Right now, before the ajax call is made, we are showing this screen: 

<img width="1095" alt="image" src="https://github.com/discourse/discourse/assets/368961/588bc8c4-a8cf-4d48-a83b-0224b1388d01">

We should show a loading indicator instead of "There are no categories matching..." 

(To get that screenshot, I had to add throttling at "Slow 3G", otherwise the local Ajax request was happening too fast.)